src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4339,38 +4137,33 @@ class MavlinkStreamNavControllerOutput : public MavlinkStream
 
 	unsigned get_size() override
 	{
-		return (_pos_ctrl_status_sub->is_published()) ?
+		return (_pos_ctrl_status_sub.advertised()) ?
 		       MAVLINK_MSG_ID_NAV_CONTROLLER_OUTPUT_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES : 0;
 	}
 
 private:
-	MavlinkOrbSubscription *_pos_ctrl_status_sub;
-	MavlinkOrbSubscription *_tecs_status_sub;
-
-	uint64_t _pos_ctrl_status_timestamp{0};
-	uint64_t _tecs_status_timestamp{0};
+	uORB::Subscription _pos_ctrl_status_sub{ORB_ID(position_controller_status)};
+	uORB::Subscription _tecs_status_sub{ORB_ID(tecs_status)};
 
 	/* do not allow top copying this class */
 	MavlinkStreamNavControllerOutput(MavlinkStreamNavControllerOutput &) = delete;
 	MavlinkStreamNavControllerOutput &operator = (const MavlinkStreamNavControllerOutput &) = delete;
 
 protected:
-	explicit MavlinkStreamNavControllerOutput(Mavlink *mavlink) : MavlinkStream(mavlink),
-		_pos_ctrl_status_sub(_mavlink->add_orb_subscription(ORB_ID(position_controller_status))),
-		_tecs_status_sub(_mavlink->add_orb_subscription(ORB_ID(tecs_status)))
+	explicit MavlinkStreamNavControllerOutput(Mavlink *mavlink) : MavlinkStream(mavlink)
 	{}
 
 	bool send(const hrt_abstime t) override
 	{
-		position_controller_status_s pos_ctrl_status = {};
-		tecs_status_s tecs_status = {};
+		if (_pos_ctrl_status_sub.updated()) {
```

Comment:
I suppose this change of functionality is intended? It is now only triggered on `pos_ctrl_status` updated and not on `tecs_status` anymore.

Reply:
Yes, the MC position controller publishes a (partially populated) `position_controller_status` now.